cmd/juju/model/exportbundle.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package model
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+
+	"github.com/juju/juju/api/bundle"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/cmd/output"
+)
+
+// NewExportBundleCommand returns a fully constructed export bundle command.
+func NewExportBundleCommand() cmd.Command {
+	return modelcmd.Wrap(&exportBundleCommand{})
+}
+
+type exportBundleCommand struct {
+	modelcmd.ModelCommandBase
+	out cmd.Output
+	api ExportBundleModelAPI
+	// name of the charm bundle file.
+	Filename string
+}
+
+const exportBundleHelpDoc = `
+Exports the current model configuration into a YAML file.
+
+Examples:
+
+    juju export-bundle
+
+If --filename is not used, the bundle is displayed in stdout.
+`
+
+// Info implements Command.
+func (c *exportBundleCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "export-bundle",
+		Purpose: "Exports the current model configuration in a charm bundle.",
+		Doc:     exportBundleHelpDoc,
+	}
+}
+
+// SetFlags implements Command.
+func (c *exportBundleCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	c.out.AddFlags(f, "yaml", output.DefaultFormatters)
+	f.StringVar(&c.Filename, "filename", "", "Export Model")
+}
+
+// Init implements Command.
+func (c *exportBundleCommand) Init(args []string) error {
+	return cmd.CheckEmpty(args)
+}
+
+// ExportBundleAPI specifies the used function calls of the ModelManager.
+type ExportBundleModelAPI interface {
```

Comment:
Comment does not match the actual name nor the actual purpose. ModelManager has nothing to do here... cut-and-paste error maybe? Also I think that ExportBundleAPI makes more sense.

Reply:
Done